src/utils/c_utils/src/string_c_utils.c

```diff
@@ -422,50 +422,50 @@ bool MallocAndSubstr(char** target, char* source, size_t len)
  * @param src The source string to be copied.
  * @param destByteLen The size of the destination buffer in bytes, including plus 1 for the null terminator.
  * @param numSrcCharsToCopy The number of source chars to copy. It will be truncated and null-terminated if >= destByteLen.
- * @return int zero on success, or a negative number if there was an error.
+ * @return true on success.
  */
-int ADUC_Safe_StrCopyN(char* dest, const char* src, size_t destByteLen, size_t numSrcCharsToCopy)
+bool ADUC_Safe_StrCopyN(char* dest, const char* src, size_t destByteLen, size_t numSrcCharsToCopy)
 {
     size_t srcLen = 0;
 
     if (dest == NULL || src == NULL || destByteLen == 0)
     {
-        return -1;
+        return false;
     }
     *dest = '\0';
 
     if (numSrcCharsToCopy >= destByteLen)
     {
-        return -2;
+        return false;
     }
 
     srcLen = ADUC_StrNLen(src, numSrcCharsToCopy);
```

Comment:
nit (minor optimization): could just start copying, and fail if we get to srcLen characters.